lib/adapter.js

```diff
@@ -0,0 +1,548 @@
+// Copyright IBM Corp. 2014,2016. All Rights Reserved.
+// Node module: strong-supervisor
+// This file is licensed under the Artistic License 2.0.
+// License text available at https://opensource.org/licenses/Artistic-2.0
+
+'use strict';
+
+var EventEmitter = require('events').EventEmitter;
+var assert = require('assert');
+var appmetrics; // = require('appmetrics') if started
+var appmetricsStarted = false; // XXX(sam) always === !!appmetrics, can remove?
+var debug = require('./debug')('adapter');
+var monitor;
+
+// Export an EventEmitter
+
+module.exports = exports = new EventEmitter;
+
+// Export the desired functions
+var metrics = {
+  startCpuProfiling: startCpuProfiling,
+  stopCpuProfiling: stopCpuProfiling,
+  startTrackingObjects: startTrackingObjects,
+  stopTrackingObjects: stopTrackingObjects
+};
+
+var internal = {
+  on: internal_on, // XXX(sam) where is this used?
+  supports: {watchdog: false} // XXX(sam) probably not needed
+};
+
+var dyninst = {
+  metrics: false
+};
+
+exports.use = use;
+exports.metrics = metrics;
+exports.profile = profile;
+exports.start = start;
+exports.internal = internal;
+exports.dyninst = dyninst;
+exports.lrtime = lrtime;
+
+
+// Global objects to store metric data
+var eventLoopMetrics = {
+  count: 0,
+  minimum: 0,
+  maximum: 0,
+  average: 0,
+};
+var cpuMetrics = {
+  system: 0,
+  user: 0,
+  count: 0,
+};
+var httpMetrics = {
+  count: 0,
+};
+var probeList = [
+  'leveldown',
+  'memcached',
+  'mongo',
+  'mysql',
+  'oracle',
+  'oracledb',
+  'postgres',
+  'redis',
+  'socketio',
+  'strong-oracle',
+  'express',
+];
+var probeMetrics = {};
+var heapMetrics = {
+  total: 0,
+  used: 0,
+  count: 0,
+};
+var gcMetrics = {
+  used: 0,
+  count: 0,
+};
+var messageMetrics = {
+  sent: 0,
+  received: 0,
+};
+
+// Keep track of Cpu Profiling
+var profiling = false;
+var objectTrackingEnabled = false;
+var lastObjectHistogram = {};
+
+// Callback interval
+var callbackInterval = process.env.STRONGLOOP_BASE_INTERVAL || 60000;
```

Comment:
If you're calling the other one `objectTrackingCallbackInterval`, shouldn't this one be `baseCallbackInterval`?